corehq/apps/userreports/reports/data_source.py

```diff
@@ -66,7 +66,10 @@ def table_name(self):
 
     @property
     def filters(self):
-        return [fv.to_sql_filter() for fv in self._filter_values.values()]
+        return filter(
+            lambda sql_filter: sql_filter is not None,
+            [fv.to_sql_filter() for fv in self._filter_values.values()]
+        )
```

Comment:
If you just pass `None` to `filter`, it will remove all falsy values:
```python
In [1]: filter(None, [0, "hi", False, 43, None])
Out[1]: ['hi', 43]
```
Unless you're trying to keep some falsy, but `not None` values.